samples/snippets/csharp/VS_Snippets_Misc/cancellation/cs/cancellationex11.cs

```diff
@@ -41,23 +41,22 @@ static void Main()
    static void NestedLoops(Rectangle rect, CancellationToken token)
    {
       for (int x = 0; x < rect.columns && !token.IsCancellationRequested; x++) {
+         // Assume that we know that the inner loop is very fast.
+         // Therefore, checking once per row in the outer loop is sufficient.
          for (int y = 0; y < rect.rows; y++) {
             // Simulating work.
             Thread.SpinWait(5000);
             Console.Write("{0},{1} ", x, y);
          }
+      }
 
-         // Assume that we know that the inner loop is very fast.
-         // Therefore, checking once per row is sufficient.
-         if (token.IsCancellationRequested) {
-            // Cleanup or undo here if necessary...
-            Console.WriteLine("\r\nCancelling after row {0}.", x);
-            Console.WriteLine("Press any key to exit.");
-            // then...
-            break;
-            // ...or, if using Task:
-            // token.ThrowIfCancellationRequested();
-         }
+      if (token.IsCancellationRequested) {
+         // Cleanup or undo here if necessary...
+         Console.WriteLine("\r\nCancelling after row {0}.", x);
```

Comment:
Is `x` in scope here?

Reply:
I don't see how cancellation is useful after the heavy work is already done?

Reply:
true, `x` is not in scope, I'll fix that.

As the usefulness of the cancellation is concerned, the overall behavior of the code did not change.

The polling example's main part is the `IsCancellationRequested` check in the loop condition, which did not change.

I suppose you could remove the additional `if` condition, which is only there to provide a place for clean up code in case of cancellation.

I am not sure what you mean by "heavy work is already done" as the outer loop can be left at any point when cancellation is requested.

Reply:
I could be misunderstanding something (and in fact I'm not sure what's the issue being resolved here), but I see `token.IsCancellationRequested` is done after the loops have already finished and completed the work.

This also doesn't make sense for me with the existing comment that says:

>         // Assume that we know that the inner loop is very fast.
>         // Therefore, checking once per row in the outer loop is sufficient.

What do you think @BillWagner?

Reply:
I updated the scope, the comments and some other issues with mixed up columns/rows.

`token.IsCancellationRequested` is called _in the loop condition_ (so polling each column), not only after the loops. So the loops may not have completed their work. As I understand the original example, the call to `token.IsCancellationRequested` after the loops is only for clean up or calling `token.ThrowIfCancellationRequested`.

As the linked issue correctly states, there is an unexpected behavior if the cancellation is requested _after_ the `if` condition is evaluated and _before_ the loop condition is evaluated. In this case, the clean up code is not called.

Reply:
> token.IsCancellationRequested is called in the loop condition

Ah I missed that.